pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -1051,13 +1052,13 @@ func buildGatewayVirtualHostDomains(hostname string, port int) []string {
 // Invalid cipher suites lead Envoy to NACKing. This filters the list down to just the supported set.
 func filteredCipherSuites(server *networking.Server) []string {
 	suites := server.Tls.CipherSuites
-	ret := make([]string, 0, len(suites))
+	ret := sets.NewSet()
 	for _, s := range suites {
-		if security.IsValidCipherSuite(s) {
-			ret = append(ret, s)
+		if security.IsValidCipherSuite(s) && !ret.Contains(s) {
+			ret = ret.Insert(s)
 		} else if log.DebugEnabled() {
-			log.Debugf("ignoring unsupported cipherSuite: %q for server %s", s, server.String())
+			log.Debugf("ignoring unsupported/duplicated cipherSuite: %q for server %s", s, server.String())
```

Comment:
can we have a different error for each case? May cause confusion as is

Reply:
Done